samples/seat_massager/consumer/src/main.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+mod consumer_impl;
+
+use digital_twin_model::{sdv_v1 as sdv, Metadata};
+use env_logger::{Builder, Target};
+use log::{debug, info, LevelFilter, warn};
+use samples_common::constants::{digital_twin_operation, digital_twin_protocol};
+use samples_common::utils::{discover_digital_twin_provider_using_ibeji, retrieve_invehicle_digital_twin_url};
+use samples_common::consumer_config;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_consumer::digital_twin_consumer_server::DigitalTwinConsumerServer;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_provider::digital_twin_provider_client::DigitalTwinProviderClient;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_provider::{GetRequest, SetRequest};
+use serde_derive::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tokio::time::{sleep, Duration};
+use tonic::transport::Server;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Property {
+    #[serde(rename = "MassageAirbags")]
+    massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+/// Start the seat massage sequence.
+///
+/// # Arguments
+/// `provider_uri` - The provider uri.
+fn start_seat_massage_sequence(provider_uri: String) {
+    debug!("Starting the consumer's seat massage sequence.");
+
+    let mut crest_row: i8 = 0;
+    let mut is_wave_moving_forwards = true;
+    const MAX_ROW: i8 = 5;
+
+    let metadata: Metadata =
+        Metadata { model: sdv::airbag_seat_massager::massage_airbags::ID.to_string() };
+    let mut property: Property = Property { massage_airbags: Vec::new(), metadata };
+
+    tokio::spawn(async move {
+        loop {
+            // We assume that the seat has 5 rows with 2 airbags in each row.
+            // The sequence will represent a wave with the crest_row representing the crest of the wave.
+            property.massage_airbags = Vec::new();
+            for airbag in 0..18 {
+                let row = airbag / 3;
+                let rows_from_crest = i8::abs(crest_row - row);
+                let mut inflation_level = 100 - (rows_from_crest * 20);
+                if inflation_level < 0 {
```

Comment:
you can also use the `max` function instead of an if statement:

```rust
use std::cmp::max;
let mut inflation_level = max(100 - (rows_from_crest * 20), 0);
```

Reply:
I will use this.